targets/Program.cs

```diff
@@ -0,0 +1,112 @@
+[assembly: AssemblyInformationalVersion(""{assemblyInformationalVersion}"")]
+";
+                if (!File.Exists(versionInfoFile) || versionContents != File.ReadAllText(versionInfoFile, Encoding.UTF8))
+                {
+                    File.WriteAllText(versionInfoFile, versionContents, Encoding.UTF8);
+                }
+            });
+
+        Targets.Add(
+            "build",
+            DependsOn("versionInfoFile", "logsDirectory"),
+            () => Run(
+                "dotnet",
+                $"build {solutionFile} /p:Configuration=Release /nologo /m /v:m " +
+                    $"/fl /flp:LogFile={logsDirectory}/build.log;Verbosity=Detailed;PerformanceSummary"));
+
+        Targets.Add(
+            "pack",
+            DependsOn("build", "outputDirectory", "readVersion"),
+            () =>
+            {
+                Run("dotnet", $"pack {mainProjectFile} --configuration Release --no-build --output {outputDirectory} /p:Version={version}");
+            });
+
+        Targets.Add(
+            "test",
+            DependsOn("build", "testsDirectory"),
+            () =>
+            {
+                foreach (var testProjectDirectory in testProjectDirectories)
+                {
+                    var outputBase = Path.GetFullPath(Path.Combine(testsDirectory, Path.GetFileName(testProjectDirectory)));
+                    Run("dotnet", $"xunit -configuration Release -nologo -xml {outputBase}.xml -html {outputBase}.html", testProjectDirectory);
+                }
+            });
+
+        Targets.Add(
+            "readVersion", () =>
+            {
+                var versionFromReleaseNotes = File.ReadLines(releaseNotesFile, Encoding.UTF8)
+                    .First(line => line.StartsWith("## ", StringComparison.Ordinal)).Substring(3).Trim();
+                Console.WriteLine($"Read version '{versionFromReleaseNotes}' from release notes");
+                var tagName = Environment.GetEnvironmentVariable("APPVEYOR_REPO_TAG_NAME");
+                if (versionFromReleaseNotes != tagName)
+                {
+                    Console.WriteLine($"Release notes version does not match tag name '{tagName}'. Disambiguating.");
+                    if (!versionFromReleaseNotes.Contains('-', StringComparison.Ordinal))
+                    {
+                        versionFromReleaseNotes += "-adhoc";
+                    }
+
+                    version = versionFromReleaseNotes +
+                        "+Build." + (Environment.GetEnvironmentVariable("APPVEYOR_BUILD_NUMBER") ?? "adhoc") +
+                        "-Sha." + (Environment.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT") ?? "adhoc");
+                }
+                else
+                {
+                    version = versionFromReleaseNotes;
+                }
+            });
+
+        Targets.Run(args);
+    }
+
+    private static string[] DependsOn(params string[] dependencies) => Targets.DependsOn(dependencies);

```

Comment:
I was kind of hoping people wouldn't feel the need to do this. It's a bit of a smell from the API... 💭

Reply:
It keeps me from accidentally forgetting `Targets`. And if you end up changing the API like you mentioned in the FakeItEasy conversion method (to `AddTarget`, `RunTargets`, …), I can move back to just a `using static` and elide the method.

Reply:
https://github.com/adamralph/bullseye/issues/34